### PR TITLE
Check for Username instead of Name for root

### PR DESCRIPTION
**Pull request by MonoAnji**
for "--vm-driver=none":
user.Name may be empty as per https://golang.org/pkg/os/user/#User which happens to be true on Manjaro 18.0.4 for root. Actually checking for the Username instead (display-) Name seems to be more reasonable.

### CODE REVIEW

**Comment by medyagh** [comment]
@MonoAnji Thank you for this contribution ! great catch, this will improve minikube's checks !

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -471,9 +471,9 @@ func validateUser() {
 	d := viper.GetString(vmDriver)
 	// Check if minikube needs to run with sudo or not.
 	if err == nil {
-		if d == constants.DriverNone && u.Name != "root" {
+		if d == constants.DriverNone && u.Username != "root" {
 			exit.UsageT(`Please run with sudo. the vm-driver "{{.driver_name}}" requires sudo.`, out.V{"driver_name": constants.DriverNone})
-		} else if u.Name == "root" && !(d == constants.DriverHyperv || d == constants.DriverNone) {
+		} else if u.Username == "root" && !(d == constants.DriverHyperv || d == constants.DriverNone) {
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: Thank you! This should really be checking for `u.Uid == 0`, as systems can have multiple root-access accounts with a uid of 0.

The root user can also be renamed.

**Comment by blueelvis** [comment]
@tstromberg @medyagh - Just FYI, this is being implemented as part of this - https://github.com/kubernetes/minikube/pull/4745

**Comment by RA489** [comment]
@minikube-bot OK to test

**Comment by medyagh** [comment]
> @tstromberg @medyagh - Just FYI, this is being implemented as part of this - #4745

I am closing this in favor of  https://github.com/kubernetes/minikube/pull/4745

**Pull request closed without merging**